pycon/finaid/views.py

```diff
@@ -425,3 +425,33 @@ def get_value(name, object):
         writer.writerow(data)
 
     return response
+
+
+@login_required
+def receipt_form(request):
+    receipts = Receipt.objects.filter(user__username=request.user.username)
+
+    if request.method == 'POST':
+        form = ReceiptForm(request.POST, request.FILES)
+
+        if form.is_valid():
+            form.instance.user = request.user
+            form.instance.application = request.user.financial_aid
+            form.save()
+
+            # Display a message to user
+            messages.add_message(request, messages.INFO,
+                                _(u"Receipt submitted"))
+
```

Comment:
After successfully processing a form POST, the pattern we follow almost 100% of the time is to redirect to another page. That way, the browser ends up by GETting a page, and if the user then hits Reload, they don't accidentally process the form again.

The one question is, which page to redirect to?  In this case, I'm guessing they might want to upload another form, so redirecting to this same page would be fine.